test/typings/either-spec.ts

```diff
@@ -75,3 +75,24 @@ console.assert(twelve.right() === 12);
 console.assert(oops.left() === "oops");
 
 console.log(nameError, messageCopy);
+
+function noThrow<E>(): Either<Error | E, string> {
+    return Either.attempt(() => "yay");
+}
+
+const e = new Error("Some error");
+const eObj = {err: "Some error"};
+function catchError<E>(): Either<Error | E, string> {
+    return Either.attempt(() => {throw e; });
+}
+
+function catchErrorObj<E>(): Either<Error | E, string> {
```

Comment:
I see only `Either<Error | X, …` examples here. But it's also important to test how TS handles to infer these type params, and how does it fit for  simple types beeing thrown..